docs/GettingStartedDocs/Contributors/SignUsingEngines.md

```diff
@@ -0,0 +1,114 @@
+# Signing a Build With OpenSSL Engines
+
+OpenSSL allows extensions called "engines" to perform the generation of keys, signing and encryption.
+Engines are useful. They can be used to allow hardware offload of encryption duties, or improved 
+encapsulation of secrets. Use of engines is required in some build systems as a way to avoid the
+storage of keys in plain text .pem files.
+
+oesign has been equipped with the ability to specify engines in the enclave signing process. In order
+to use the engine, the command line syntax has been extended.
+```
+oesign sign --enclave-image ENCLAVE_IMAGE --config-file CONFIG_FILE --engine engine-NAME -load-path engine-LOAD-PATH-key KEY-ID
+
+```
+
+## Engines 
+
+The concept of an engine was added to OpenSSL starting with 0.9.6. There are built in engine implementations
+implementations for
+- Microsoft CryptoAPI
+- VIA Padlock
+- nCipher CHIL
+and a api for controlling dynamically bound engines which are bound from shared libraries at runtime via either program logic
+or configuration files. 
+
+The openssl commnd line utility can show available engines via the "engine" command.
+```
+# openssl engine
+(rdrand) Intel RDRAND engine
+(dynamic) Dynamic engine loading support
+
+```
+Of particular interest is the "dynamic" engine, which allow dynamic plugins to perform encryption duties. 
+These objects allow a plugin mechanism for implementations of cryptographic algorithms and can be statically linked in the
+OpenSSL library at compile-time or dynamically loaded at run-time in and out of the running application 
+with low overhead from external binary shared objects implementing the engine API using special built-in engine
```

Comment:
s/using special engine called "dynamic"/using the dynamic engine/

Rationale: (1) missing article.  (2) "dynamic" engine was already introduced in the first sentence of the paragraph

Reply:
Dynamic engines is the term of art for .so engines which are dynalinked by the engine named "dynamic".
I'm not thrilled with the terminology either, but if I try to define new terms it will confuse folks who are used to the openssl documentation.

Proposed revision:
Of particular interest is the "dynamic" engine, which allow dynamic plugins (dynamic engines) to perform encryption duties.
These objects allow a plugin mechanism for implementations of cryptographic algorithms and can be statically linked in the
OpenSSL library at compile-time or dynamically loaded at run-time in and out of the running application
with low overhead from external binary shared objects implementing the engine API using special built-in engine
called “dynamic” to perform the loading.

Reply:
I now understand you're using "dynamic engine" and "dynamic plugin" as synonyms, instead of plugins being things that plug into an engine.   As such, proposed revision:

Of particular interest are "dynamic" engines, which are plugins that perform encryption duties. These plugins for implementations of cryptographic algorithms can be statically linked in the
OpenSSL library at compile-time or dynamically loaded at run-time in and out of the running application with low overhead from external binary shared objects implementing the engine API.

Reply:
I do not believe dynamic engines are ever statically linked,

Reply:
not, dynamic engine is a term of art, but dynamic plugin is a description, not a synonym. This is necessary because all openssl documentation uses that terminology.

Reply:
I rewrote the text to try to make it more clear.